tests/functionaltests/Tests/CortanaTests.mm

```diff
@@ -0,0 +1,103 @@
+
+           public:
+               MOCK_STDCALL_METHOD_1(get_Status);
+               MOCK_STDCALL_METHOD_1(get_Text);
+               MOCK_STDCALL_METHOD_1(get_Confidence);
+               MOCK_STDCALL_METHOD_1(get_SemanticInterpretation);
+               MOCK_STDCALL_METHOD_2(GetAlternates);
+               MOCK_STDCALL_METHOD_1(get_Constraint);
+               MOCK_STDCALL_METHOD_1(get_RulePath);
+               MOCK_STDCALL_METHOD_1(get_RawConfidence);
+               MOCK_STDCALL_METHOD_1(get_PhraseStartTime);
+               MOCK_STDCALL_METHOD_1(get_PhraseDuration);
+           });
+
+MOCK_CLASS(
+    MockVoiceCommandActivatedEventArgs,
+    public RuntimeClass<RuntimeClassFlags<WinRtClassicComMix>, IVoiceCommandActivatedEventArgs, IActivatedEventArgs> {
+        InspectableClass(RuntimeClass_Windows_ApplicationModel_Activation_VoiceCommandActivatedEventArgs,
+                         BaseTrust); // claim to be the implementation for the real system RuntimeClass for VoiceCommandActivatedEventArgs.
+
+    public:
+        MOCK_STDCALL_METHOD_1(get_Result);
+        MOCK_STDCALL_METHOD_1(get_Kind);
+        MOCK_STDCALL_METHOD_1(get_PreviousExecutionState);
+        MOCK_STDCALL_METHOD_1(get_SplashScreen);
+    });
+
+MOCK_CLASS(MockProtocolActivatedEventArgs,
+           public RuntimeClass<RuntimeClassFlags<WinRtClassicComMix>, IProtocolActivatedEventArgs, IActivatedEventArgs> {
+               InspectableClass(RuntimeClass_Windows_ApplicationModel_Activation_ProtocolActivatedEventArgs,
+                                BaseTrust); // claim to be the implementation for the real system RuntimeClass for Accel.
+
+           public:
+               MOCK_STDCALL_METHOD_1(get_Uri);
+               MOCK_STDCALL_METHOD_1(get_Kind);
+               MOCK_STDCALL_METHOD_1(get_PreviousExecutionState);
+               MOCK_STDCALL_METHOD_1(get_SplashScreen);
+           });
+
+// TODO: 8008558 Test handling foreground activation parameters from Cortana
+TEST(CortanaTest, VoiceCommandForegroundActivation) {
+    LOG_INFO("CortanaTest Voice Command Foreground Activation Test: ");
+    // Get current application to test on
+    WXApplication* app = [WXApplication current];
+    // Create mocked data to pass into application
+    auto fakeSpeechRecognitionResult = Make<MockSpeechRecognitionResult>();
+    fakeSpeechRecognitionResult->Setget_Text([](HSTRING* text) {
+        Wrappers::HString value;
+        value.Set(L"CORTANA_TEST");
+        *text = value.Detach();
+        return S_OK;
+    });
+    auto fakeVoiceCommandActivatedEventArgs = Make<MockVoiceCommandActivatedEventArgs>();
+    fakeVoiceCommandActivatedEventArgs->Setget_Result([&fakeSpeechRecognitionResult](ISpeechRecognitionResult** result) {
+        fakeSpeechRecognitionResult.CopyTo(result);
+        return S_OK;
+    });
+    fakeVoiceCommandActivatedEventArgs->Setget_Kind([](ActivationKind* kind) {
+        *kind = ActivationKind_VoiceCommand;
+        return S_OK;
+    });
```

Comment:
nit: coding conventions state that a new line should be present after a closing } block. This makes each {} section a discrete block and more easily read.